vm/devices/vmbus/vmbus_server/src/channels.rs

```diff
@@ -1442,6 +1457,7 @@ impl<'a, N: 'a + Notifier> ServerWithNotifier<'a, N> {
                 } => {
                     send_open_result(
                         self.notifier,
+                        &mut self.inner.pending_messages,
```

Comment:
I wonder if you could make all this sending stuff easier to read with something like:

```rust
struct PendingMessages(VecDeque<..>);
struct MessageSender<'a, N>(&mut PendingMessages, &mut N);
impl PendingMessages {
    fn sender(&mut self, notifier: &mut N) -> MessageSender<'a, N> {..}
}
impl<N:..> MessageSender<'_, N> {
    fn send_open_result(&mut self, ..) { .. }
    ..
}
```

This might make it easier to pass the `(notifier, pending_messages)` pair around, as well as to group all the message send functions in one place.